config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.groovy

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.config.annotation.web.configurers.oauth2.client
+
+import org.springframework.context.annotation.Bean
+import org.springframework.security.authentication.ProviderManager
+import org.springframework.security.config.annotation.BaseSpringSpec
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.config.oauth2.client.CommonOAuth2Provider
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.AuthorityUtils
+import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper
+import org.springframework.security.oauth2.client.authentication.OAuth2LoginAuthenticationProvider
+import org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter
+
+/**
+ * test for OAuth2LoginConfigurer.
+ * @author Kazuki Shimizu
+ */
+class OAuth2LoginConfigurerTests extends BaseSpringSpec {
```

Comment:
Can you please convert this to JUnit. We're in the process of converting all our tests to JUnit as we're discontinuing Groovy.

Reply:
of course!

Reply:
I've fixed via b876aada8c326d6cd8d59e8db8a075fec631ac86.